injectproxy/routes.go

```diff
@@ -41,11 +67,20 @@ func NewRoutes(upstream *url.URL, label string) *routes {
 		label:    label,
 	}
 	mux := http.NewServeMux()
-	mux.Handle("/federate", enforceMethods(r.federate, "GET"))
+	mux.Handle("/federate", enforceMethods(r.matcher, "GET"))
 	mux.Handle("/api/v1/query", enforceMethods(r.query, "GET", "POST"))
 	mux.Handle("/api/v1/query_range", enforceMethods(r.query, "GET", "POST"))
 	mux.Handle("/api/v1/alerts", enforceMethods(r.noop, "GET"))
 	mux.Handle("/api/v1/rules", enforceMethods(r.noop, "GET"))
+	mux.Handle("/api/v1/series", enforceMethods(r.matcher, "GET"))
```

Comment:
The `/federate` endpoint of the proxy throws away any existing `match[]` field in the incoming query and the outgoing query is just `match[]={<label>="<label value>"}`.

This doesn't really line up with the README.md that says:

> Similar to query endpoint, for metadata endpoints `/api/v1/series`, `/api/v1/labels`, `/api/v1/label/<name>/values` the proxy inject the specified label in the `match` selector.

Reply:
Very good point, thanks for spotting. Let me change the PR to match docs and let's discuss. I think adding new `match[]` is what we want

Reply:
actually ignored me, we need to either select within query or replace indeed